Allow overriding theme config

blackdaemon opened the pull request:
Currently if theme is used, it's impossible to override its settings in the setup().
Merging the theme config with user config allows to override specific fields in the theme config.

blackdaemon committed:
Merge theme config with user config

blackdaemon commented:
I squashed the commits into one

imsi32 commented:
There were 2 issues that requested this feature but I considered as it can break users theme and also the way  I think to achieve this feature was slow in https://github.com/imsi32/yatline.yazi/issues/41 and https://github.com/imsi32/yatline.yazi/issues/38. However, thanks to you, I find an efficient way to implement this feature. So it will merged.

Thank you.

imsi32 committed:
[FEAT] Overriding Theme

details: Ability to overriding theme in config added. Also config removed after it gives its value to global variables to free memory.
